gists/gist7480523/main.go

```diff
@@ -80,18 +81,33 @@ func (this *GoPackage) UpdateVcs() {
 }
 
 func (this *GoPackage) UpdateVcsFields() {
-	if this.Dir.Repo != nil {
-		MakeUpdated(this.Dir.Repo.VcsLocal)
-		MakeUpdated(this.Dir.Repo.VcsRemote)
+	if this.Dir.Repo == nil {
+		return
+	}
 
-		repoImportPath := GetRepoImportPath(this.Dir.Repo.Vcs.RootPath(), this.Bpkg.SrcRoot)
-		if repoRoot, err := vcs.RepoRootForImportPath(repoImportPath, false); err == nil {
-			this.Dir.Repo.RepoRoot = repoRoot
-		}
+	MakeUpdated(this.Dir.Repo.VcsLocal)
+	MakeUpdated(this.Dir.Repo.VcsRemote)
+
+	repoImportPath := GetRepoImportPath(this.Dir.Repo.Vcs.RootPath(), this.Bpkg.SrcRoot)
+	if repoRoot, err := vcs.RepoRootForImportPath(repoImportPath, false); err == nil {
+		this.Dir.Repo.RepoRoot = repoRoot
 	}
 }
 
-func GetRepoImportPath(repoPath, srcRoot string) (repoImportPath string) {
+func GetRepoImportPath(repoPath, srcRoot string) string {
+	s, err := filepath.EvalSymlinks(repoPath)
+	if err != nil {
+		fmt.Fprintln(os.Stderr, "warning: GetRepoImportPath: can't resolve symlink")
+	} else {
+		repoPath = s
+	}
```

Comment:
Since the value of `s` is only needed within the scope of this `if` block, I think it would be cleaner to write this as:

```Go
if s, err := filepath.EvalSymlinks(repoPath); err == nil {
	repoPath = s
} else {
	fmt.Fprintln(os.Stderr, "warning: GetRepoImportPath: can't resolve symlink")
}
```

(I've also swapped the case order, since the error is not fatal and does result in `return`ing early, I put it in the `else` block.)